source/common/stats/stats_impl.cc

```diff
@@ -135,35 +135,47 @@ bool TagExtractorImpl::extractTag(const std::string& stat_name, std::vector<Tag>
   return false;
 }
 
-RawStatData* HeapRawStatDataAllocator::alloc(const std::string& name) {
-  uint64_t num_bytes_to_allocate = RawStatData::structSize(name.size());
-  RawStatData* data = static_cast<RawStatData*>(::calloc(num_bytes_to_allocate, 1));
-  if (data == nullptr) {
-    throw std::bad_alloc();
-  }
-  data->checkAndInit(name, num_bytes_to_allocate);
```

Comment:
If we're getting rid of checkAndInit, is there other cleanup in the regular RawStatData struct we can do now, now that it's been decoupled from HeapStatData? Maybe RawStatData, if only for blocked / shared memory, can be less general and more opinionated about its internal memory representation.

Reply:
I don't think we're getting rid of checkAndInit. It is still needed in the shm case. I don't think anything gets easier by knowing it's always shm.

Reply:
correction...checkAndInit is just merged with initialize() now.